python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1158 @@
+            for axis in constant_axes:
+                input = bb.emit_te(topi.expand_dims, input, axis=axis, num_newaxis=1)
+            return input
+
+        raise NotImplementedError("Unsqueeze with dynamic axes is not supported.")
+
+
+class Concat(OnnxOpConverter):
+    """Convert an onnx Concat node into an equivalent Relax expression."""
+
+    @classmethod
+    def _impl_v13(cls, bb, inputs, attr):
+        axis = attr.get("axis", 0)
+        return bb.emit_te(topi.concatenate, inputs, axis)
+
+
+class Add(OnnxOpConverter):
+    """Convert an onnx Add node into an equivalent Relax expression."""
+
+    @classmethod
+    def _impl_v13(cls, bb, inputs, attr):
+        return bb.emit_te(topi.add, inputs[0], inputs[1])
+
+
+class Mul(OnnxOpConverter):
+    """Convert an onnx Mul node into an equivalent Relax expression."""
+
+    @classmethod
+    def _impl_v13(cls, bb, inputs, attr):
+        return bb.emit_te(topi.multiply, inputs[0], inputs[1])
+
+
+class Cast(OnnxOpConverter):
+    """Convert an onnx Cast node into an equivalent Relax expression."""
+
+    @classmethod
+    def _impl_v13(cls, bb, inputs, attr):
+        to_type = get_type(attr["to"])
+        return bb.emit_te(topi.cast, inputs[0], to_type)
+
+
+class Gather(OnnxOpConverter):
+    """Convert an onnx Gather node into an equivalent Relax expression."""
+
+    @classmethod
+    def _impl_v13(cls, bb, inputs, attr):
+        # TODO This assumes positive only indices.
+        axis = attr.get("axis", 0)
+        return bb.emit_te(topi.take, inputs[0], inputs[1], axis)
+
+
+class Gemm(OnnxOpConverter):
+    """Convert an onnx Gemm node into an equivalent Relax expression."""
+
+    @classmethod
+    def _impl_v13(cls, bb, inputs, attr):
+        alpha = attr.get("alpha", None)
+        beta = attr.get("beta", None)
+        transA = attr.get("transA", False)
+        transB = attr.get("transB", False)
+        A = inputs[0]
+        B = inputs[1]
+        C = inputs[2]
+        dtype = A.checked_type.dtype
+
+        # Compute Y = alpha * A X B + beta * C
+
+        if alpha is not None:
+            A = bb.emit_te(topi.multiply, A, relax.const(alpha, dtype=dtype))
+
+        Y = bb.emit_te(topi.matmul, A, B, transA, transB)
+
+        if C is not None:
+            if beta is not None:
+                C = bb.emit_te(topi.multiply, C, relax.const(beta, dtype=dtype))
+            Y = bb.emit_te(topi.add, Y, C)
+
+        return Y
+
+
+class Reshape(OnnxOpConverter):
+    """Convert an onnx Reshape node into an equivalent Relax expression."""
+
+    @classmethod
+    def _impl_v13(cls, bb, inputs, attr):
+        data = inputs[0]
+        # TODO We assume new_shape is a constant, need to enable tensor input to reshape
+        # for full support.
+        if not isinstance(inputs[1], relax.Constant):
+            return inputs[0]
+        new_shape = inputs[1].data.numpy().tolist()
+
+        # Convert -1 dims in new_shape into positive equivalent.
```

Comment:
I presume at most one dimension is allowed to be -1, right? If so, that is not checked here

Reply:
I've added a check for this